lib/mix/lib/mix/tasks/new.ex

```diff
@@ -223,7 +223,7 @@ defmodule Mix.Tasks.New do
 
   ```elixir
   def deps do
-    [{:<%= @app %>, "~> 0.1.0"}]
```

Comment:
[SemVer](http://semver.org/#spec-item-4) says:

> Major version zero (0.y.z) is for initial development. Anything may change at any time. The public API should not be considered stable.

I have seen this generally interpreted as "0.x.y" means that everything can change, but bumps in y are usually bugfixes". If we use `~> 0.1`, then we are at risk of this interpretation causing trouble because of a too restrictive version. Personally, I am 👎 on this.